_stbt/logging.py

```diff
@@ -142,3 +142,13 @@ def test(level):
             ddebug(u'Prüfungs Debug-Unicode')
     for level in [0, 1, 2]:
         yield (test, level)
+
+
+def draw_on(frame, *args, **kwargs):
+    draw_sink_ref = getattr(frame, '_draw_sink', None)
+    if not draw_sink_ref:
+        return
+    draw_sink = draw_sink_ref()
+    if not draw_sink:
+        return
+    draw_sink.draw(*args, **kwargs)
```

Comment:
"A try/except block is extremely efficient if no exceptions are raised" [[1]] so it might make sense to use a `weakref.proxy` and catch `ReferenceError`? Just bikeshedding really.

[1]: https://docs.python.org/2/faq/design.html#how-fast-are-exceptions

Reply:
I expect the sink pipeline to outlive the frames, and we manage sink pipeline teardown explicitly (not via `__del__`). I wonder what is the performance impact of strong references vs. weak references. Probably not significant considering we'll only be doing it once every 16ms at most.

Reply:
Each frame keeping the sink-pipeline alive would make me nervous, particularly given the teardown issues we've had in the past.